extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/input/external/ExternalInputSpecSlicer.java

```diff
@@ -229,7 +243,16 @@ public <T> Iterator<List<T>> split(
       return Iterators.filter(
           SlicerUtils.makeSlicesDynamic(
               inputIterator,
-              item -> inputAttributeExtractor.apply(item).getSize(),
+              item -> {
+                if (null != inputFormat) {
+                  InputFileAttribute inputFileAttribute = inputAttributeExtractor.apply(item);
+                  return inputFormat.getWeightedSize(
```

Comment:
It'd be nice for callers to not need the `InputFormat` here. How about having `InputFileAttribute` be `getSize` and `getWeightedSize`?

Reply:
Good idea, fixed